### DISCUSSION TITLE
dcm 在安装 依赖dde-file-manager的应用的时候会依赖损坏

### DISCUSSION THREAD

**Discussion by Linux-qitong** [linuxdeepin/developer-center#4452] [Troubleshooting | 故障排查]
https://bbs.deepin.org/phone/zh/post/257051
原贴
dpkg: 处理软件包 udisks2 (--configure)时出错：
 已安装 udisks2 软件包 post-installation 脚本 子进程返回错误状态 1
dpkg: 依赖关系问题使得 gvfs-daemons 的配置工作不能继续：
 gvfs-daemons 依赖于 udisks2；然而：
  软件包 udisks2 尚未配置。

dpkg: 处理软件包 gvfs-daemons (--configure)时出错：
 依赖关系问题 - 仍未被配置
dpkg: 依赖关系问题使得 gvfs-backends 的配置工作不能继续：
 gvfs-backends 依赖于 gvfs-daemons (= 1.38.1.3-deepin3)；然而：
  软件包 gvfs-daemons 尚未配置。

dpkg: 处理软件包 gvfs-backends (--configure)时出错：
 依赖关系问题 - 仍未被配置
dpkg: 依赖关系问题使得 gvfs:amd64 的配置工作不能继续：
 gvfs:amd64 依赖于 gvfs-daemons (<< 1.38.1.3-deepin3.1~)；然而：
  软件包 gvfs-daemons 尚未配置。
 gvfs:amd64 依赖于 gvfs-daemons (>= 1.38.1.3-deepin3)；然而：
  软件包 gvfs-daemons 尚未配置。

dpkg: 处理软件包 gvfs:amd64 (--configure)时出错：
 依赖关系问题 - 仍未被配置
在处理时有错误发生：
 udisks2
 gvfs-daemons
 gvfs-backends
 gvfs:amd64
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by BLumia** [comment]
这个汇报还没原帖详细= =

dcm 是啥？v23 有加哪些源？依赖 dde-file-manager 的应用有例子吗（具体是在尝试哪个的时候复现的问题）？除了第一方应用外我一时没想到其它的，而第一方应用本身不应该有问题...

**Reply by Linux-qitong** [reply]
没有原贴详细，我这边有限制

**Reply by shenmo7192** [reply]
> 这个汇报还没原帖详细= =
> 
> dcm 是啥？v23 有加哪些源？依赖 dde-file-manager 的应用有例子吗（具体是在尝试哪个的时候复现的问题）？除了第一方应用外我一时没想到其它的，而第一方应用本身不应该有问题...

20兼容环境dcm

我随便找了一个应用，这个应用是  com.gitee.deepin-opensource.desktop-entry-editor ，我使用dcm repack命令进行重新打包，然后安装了此应用。然而，在安装的时候报错了

**Reply by BLumia** [reply]
> 20兼容环境dcm

cc @black-desk

**Reply by black-desk** [reply]
这个兼容环境里面有一些东西，设计上本身就是不可用的：

比如： 包里面有内核模块的，需要访问system-bus的。

~~不知道这个udisks2是不是属于这一类。~~

~~具体原因等我有空了看看。~~

确实属于这一类， 它带了一个systemd服务，并且会在postinst里面启用它。

从设计上来说这种东西不被支持。

谢谢反馈

**Reply by black-desk** [reply]
看了一下原帖 @shenmo7192 

这个容器环境里面是100%无法拿到真root的, 看起来是udisks2的postinst里面访问了sysfs下面的什么东西.

很可能没什么解决方案.

**Reply by shenmo7192** [reply]
> 看了一下原帖 @shenmo7192
> 
> 这个容器环境里面是100%无法拿到真root的, 看起来是udisks2的postinst里面访问了sysfs下面的什么东西.
> 
> 很可能没什么解决方案.

换成docker会好一些吗？podman是无法使用root的

**Reply by black-desk** [reply]
应该考虑从其他角度修

我试了一下sid, sid在box里面装这个包

虽然一样有 Failed to write ... 但是脚本并不会以失败结束。

---

update: 虽然设计上并不支持，但是我有空会看一下具体的原因。说不定有解决方案（

---

update2:

作为临时解决方案，可将容器中的 /var/lib/dpkg/info/udisks2.postinst 文件中

```sh
udevadm trigger --subsystem-match=block --action=change
```
一行，改为
```sh
udevadm trigger --subsystem-match=block --action=change || true
```

@shenmo7192

**Reply by black-desk** [reply]
@Zeno-sole 看看能不能把20上这个包的postinst改一下？
